aea/protocols/dialogue/base.py

```diff
@@ -948,6 +995,220 @@ def add_dialogue_endstate(
             self._other_initiated[end_state] += 1
 
 
+def find_caller_object(object_type: Type):
```

Comment:
How reliable is this approach?

Reply:
I don't like this solution, but it works for cases I tested.

Probably can try to find out something with a global variable like a context. probably will be a better solution.
only one point is to create  a thread local variable, but it's provided by python.